daemon/regmaild/regmaild.c

```diff
@@ -73,6 +73,52 @@ regmaildb_open(sqlite3 **Db, const char *fpath) {
 
 // standalone initialize builder
 
+static int
+emaildb_prepare_stmt(sqlite3 *db, sqlite3_stmt **stmt)
+{
+    if (sqlite3_prepare(db, "REPLACE INTO emaildb (userid, email) VALUES (lower(?),lower(?));",
+                -1, stmt, NULL) != SQLITE_OK)
+    {
+        fprintf(stderr, "Error preparing stmt: %s\n", sqlite3_errmsg(db));
+        return 0;
+    }
+    return 1;
+}
+
+static int
+emaildb_bind_args(sqlite3 *db, sqlite3_stmt *stmt, const userec_t *u)
+{
+    if (sqlite3_bind_text(stmt, 1, u->userid, -1, SQLITE_STATIC) != SQLITE_OK ||
+        sqlite3_bind_text(stmt, 2, u->email, -1, SQLITE_STATIC) != SQLITE_OK)
+    {
+        fprintf(stderr, "\nError binding args: %s\n", sqlite3_errmsg(db));
+        return 0;
+    }
+    return 1;
+}
+
+static int
+emaildb_filter_user(const userec_t *u)
+{
+    if (strlen(u->userid) < 2 || strlen(u->userid) > IDLEN)
+        return 0;
+    if (strlen(u->email) < 5)
+        return 0;
+    return 1;
```

Comment:
If all you are returning is success and failure, maybe use bool as return type explicitly?